### ISSUE TITLE
Git pull hangs forever

### ISSUE THREAD

**Issue by easeev** [microsoft/vscode-remote-release#985]
Issue Type: <b>Bug</b>

I try to do Git Pull from VSCode and it hands indefinitely. I'm using Git on WSL.

I assume there's an issue with SSH keys since that happens when I renamed the key from id_xxx to id_rsa in the ~/.ssh folder. When it had a custom name, the issue was: `git@github.com: Permission denied (publickey)`.

Git in the WSL itself works perfectly but not from VSCode.

Extension version: 0.38.10
VS Code version: Code 1.36.1 (2213894ea0415ee8c85c5eea0d0ff81ecc191529, 2019-07-08T22:59:35.033Z)
OS version: Windows_NT x64 10.0.18936
Remote OS version: Linux x64 4.19.43-microsoft-standard

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz (8 x 2112)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|23.84GB (11.67GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu-18.04|
|OS|Linux x64 4.19.43-microsoft-standard|
|CPUs|Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz (8 x 2111)|
|Memory (System)|22.35GB (21.83GB free)|
|VM|0%|
</details>

**Comment by bobaoapae** [comment]
Hi, my vscode remote ssh hangs forever on git pull without any error

**Comment by MrPointer** [comment]
For anyone else stumbling upon it, please take a look at [this discussion](https://github.com/microsoft/WSL/issues/4246) on the WSL repo.  

**TL;DR**
WSL2 has some serious issues with VPNs that set the MTU below the default 1500, e.g. 1350/1392/1300, because it always defaults to 1500, disregarding the real value set on its interface from Windows' side, as if it were stubbed.
So if you're connected over VPN and have the same behavior, please refer to the linked issue, especially the latest comments.

**Comment by simwak** [comment]
Having the same problems sometimes with Remote SSH. `git push`/ `git pull` from command line still works. Only workaround is the close VS and reopen it again. Sometimes even two times.

**Comment by altosaar** [comment]
Having the same issue

**Comment by CharlesAttend** [comment]
Still have this issue, does anyone find a solution ?

**Comment by MartinEmrich** [comment]
Same issue, even when the key was already unlocked, and the Git server is not behind a VPN tunnel...

after killing all SSH processes and trying again, it works.

It would be nice if vscode would time out on non-progressing git-via-ssh operations after a reasonable time (maybe 30 or 60s?), and also not do all at once (having a multi root workspace with >20 repositories, and vscode using 1 cpu core for each, it just melts the CPU)

**Comment by CornJojk** [comment]
Same issue with vscode and the WSL remote extension. The WSL2 bash terminal is unable to operate git successfully, I am not on a VPN, though one is installed. When the VPN is on, WSL2 simply can't connect to any internet (that's a different issue and I don't need to fix it). My .bashrc does initialize an Anaconda base environment, though I have the same problem whether or not I subsequently deactivate it. 

Initially `git fetch`, `git pull`, or `git push` resulted in a certificate failure:
`fatal: unable to access 'https://compygitrepo': server certificate verification failed. "CAfile: none CRLfile: none"`

I fixed that by taking the steps here to save and configure a certificate:
[https://stackoverflow.com/questions/67647067/git-clone-from-gitlab-fails-on-linux-while-working-in-windows-git-bash](https://stackoverflow.com/questions/67647067/git-clone-from-gitlab-fails-on-linux-while-working-in-windows-git-bash)

Now I just get the hanging behavior noted above. A WSL2 instance from outside vscode works fine.

**Comment by J-HowHuang** [comment]
Found out that VS Code had updates and told me to restart. Afterward it ask me to login to Github again, then the problem was solved.

**Comment by MartinEmrich** [comment]
I don't use github, so the problem still persists. 

One way to trigger it for shure is to start vscode before unlocking the SSH key in the agent. (Sometimes, it happens afterwards as well)

VSCode should either relay the passphrase prompt to the UI so the user can input it (`SSH_ASKPASS`*,...), oder make the SSH child process fail for non-unlocked keys (e.g. by somehow preventing interaction with /dev/tty for the ssh process?)

**Comment by CornJojk** [comment]
> Found out that VS Code had updates and told me to restart. Afterward it ask me to login to Github again, then the problem was solved.

Are you using the remote WSL extension and using git through the Linux terminal? I had no such luck.

**Comment by CornJojk** [comment]
I finally found resolution through TWO actions. 

I think for people coming here they just need to set up a credential storage or helper rather than messing with anything related to certificates. See this problem and solution: [https://stackoverflow.com/questions/66503781/git-asks-for-password-to-push-on-wsl-2-ubuntu-but-not-on-windows-why-is-that](https://stackoverflow.com/questions/66503781/git-asks-for-password-to-push-on-wsl-2-ubuntu-but-not-on-windows-why-is-that)

First step: I did eventually get an update notification which replaced one problem for another. Rather than hanging, it was now asking me for credentials every time. Then I followed the advice at the above link, except that my git-bash installation is in the AppData folder.
